tests/test_variants.py

```diff
@@ -176,7 +176,7 @@ def test_validate_spec():
     """
     spec = {
         # normal expansions
-        "foo": [2.7, 3.7, 3.8],
+        "foo": [3.7, 3.8],
```

Comment:
Let's abstract this:

```suggestion
        "foo": [1.2, 3.4],
```

I also think this test should be reworked into a small test using parameterization